File: Pascal's Triangle (print all rows)/Pascal's Triangle.py
"""
In Pascal's triangle, 
each number is the sum of the two numbers directly above it as shown:

                 1
               1   1
            1    2     1
        1      3    3      1
    1      4      6     4     1

Example 1:

Input: numRows = 5
Output: [[1],[1,1],[1,2,1],[1,3,3,1],[1,4,6,4,1]]

Example 2:

Input: numRows = 1
Output: [[1]]


"""
def generate_pascals_triangle(numRows):
    triangle = [[1]]
    for i in range(1, numRows):
        row = [1] + [triangle[i-1][j-1] + triangle[i-1][j] for j in range(1, i)] + [1]
        triangle.append(row)
    return triangle

numRows = 5
result = generate_pascals_triangle(numRows)
for row in result:
    print(row)

File: Pascal's Triangle (print all rows)/Pascal's Triangle.cpp
#include <iostream>
#include <vector>

using namespace std;

vector<vector<int>> generatePascalsTriangle(int numRows) {
    vector<vector<int>> triangle;
    triangle.push_back({1});

    for (int i = 1; i < numRows; ++i) {
        vector<int> row = {1};
        for (int j = 1; j < i; ++j) {
            row.push_back(triangle[i - 1][j - 1] + triangle[i - 1][j]);
        }
        row.push_back(1);
        triangle.push_back(row);
    }

    return triangle;
}

int main() {
    int numRows = 5;
    vector<vector<int>> result = generatePascalsTriangle(numRows);

    for (const auto& row : result) {
        for (int num : row) {
            cout << num << " ";
        }
        cout << endl;
    }

    return 0;
}
-------------------------
| Prepared by -         |
| Himel Sarder          |
| Dept. of CSE, BSFMSTU |
-------------------------
